Backpack/src/main/java/net/skyscanner/backpack/calendar2/data/CalendarMonth.kt

```diff
@@ -43,16 +45,27 @@ internal inline fun CalendarMonth(
   monthsFormatter: SimpleDateFormat,
   weekFields: WeekFields,
   selection: CalendarSelection,
+  selectionMode: CalendarParams.SelectionMode,
+  selectWholeMonthLabel: String?,
   day: (YearMonth, LocalDate) -> CalendarCell.Day,
 ): CalendarMonth {
 
   val firstDay = days.first()
+  val allowWholeMonthSelection = (selectionMode as? CalendarParams.SelectionMode.Range)?.allowSelectWholeMonth ?: false
```

Comment:
If we had a new `Month` selection mode, here would be just `selectionMode is CalendarParams.SelectionMode.Month`